File: ByteMarket/backend/src/main/java/es/grupo04/backend/dto/ProductDTO.java
package es.grupo04.backend.dto;

import java.time.LocalDate;
import java.util.List;

public record ProductDTO(
    Long id,
    String name,
    Float price,
    String category,
    String description,
    UserBasicDTO owner, //Avoid returning the complete User object
    List<String> imageUrls,
    boolean sold,
    LocalDate publishDate
){
}

File: frontend/src/app/dtos/product.dto.ts
import { UserBasicDTO } from "./user.basic.dto";

export interface ProductDTO {
    id: number;
    name: string;
    price: number;
    category: string;
    description: string;
    owner: UserBasicDTO;
    imageUrls: string[];
    thumbnail: string;
    sold: boolean;
    publishDate: Date;
}
